consensus/core/src/network/connection_monitor.rs

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{sync::Arc, time::Duration};
+
+use anemo::{types::PeerEvent, PeerId};
+use dashmap::DashMap;
+use mysten_metrics::spawn_logged_monitored_task;
+use quinn_proto::ConnectionStats;
+use tokio::{
+    sync::oneshot::{Receiver, Sender},
+    task::JoinHandle,
+    time,
+};
+
+use super::metrics::QuinnConnectionMetrics;
+
+const CONNECTION_STAT_COLLECTION_INTERVAL: Duration = Duration::from_secs(60);
+
+pub(crate) struct ConnectionMonitorHandle {
+    handle: JoinHandle<()>,
+    stop: Sender<()>,
+    // TODO: Sui will use this component eventually instead of the NW version
```

Comment:
Does this comment mean that this component will be used by state sync? And it seems `connection_statuses` is actually used.

Reply:
> My plan was to refactor and move it to a place outside of consensus after we officially switch to mysticeti

sounds good to me